valhalla/baldr/graphreader.h

```diff
@@ -821,24 +823,7 @@ class GraphReader {
    * @param tile      which tile the edge lives in, is updated if not correct
    * @return vector of incidents
    */
-  std::vector<Incident> GetIncidents(const GraphId& edge_id, const GraphTile*& tile) {
-    // if we are not doing this for any reason then bail
-    std::shared_ptr<incident_tile_t> itile;
-    if (!simulate_incidents_ || !GetGraphTile(edge_id, tile) ||
-        !tile->trafficspeed(tile->directededge(edge_id)).has_incidents ||
-        !(itile = GetIncidentTile(edge_id))) {
-      return {};
-    }
-
-    // get the range of incidents we care about and hand it back, equal_range has no lambda option
-    auto begin = std::partition_point(itile->begin(), itile->end(), [&edge_id](const Incident& i) {
-      return i.edge_index < edge_id.id(); // first one that is >= the id we want
-    });
-    auto end = std::partition_point(begin, itile->end(), [&edge_id](const Incident& i) {
-      return i.edge_index <= edge_id.id(); // first one that is > the id we want
-    });
-    return std::vector<Incident>(begin, end);
-  }
+  IncidentResult GetIncidents(const GraphId& edge_id, const GraphTile*& tile);
```

Comment:
docstring above needs to be updated, since it doesnt return a vector but rather a range and a tile